compiler/rustc_parse/src/lexer/mod.rs

```diff
@@ -491,6 +500,35 @@ impl<'a> StringReader<'a> {
         FatalError.raise()
     }
 
+    // See RFC 3101.
+    fn report_reserved_prefix(&self, start: BytePos) {
+        let prefix_span = self.mk_sp(start, self.pos);
+        let msg = format!("prefix `{}` is unknown", self.str_from_to(start, self.pos));
+
+        if self.sess.edition < Edition::Edition2021 {
```

Comment:
Shouldn't we check the edition from `prefix_span`? I think this is equivalent, but it seems more correct to me.

You can just invoke `!prefix_span.rust_2021()` I think.

(cc @petrochenkov)

Reply:
> Shouldn't we check the edition from prefix_span? I think this is equivalent

They are equivalent if `self.mk_sp` uses the root context. But this span (including context) can be overridden by `override_span`, which is used by the proc macro server to select the span (and thus edition) in which the tokens are parsed.

So yeah, this should use `.rust_2021()` to work correctly in with proc macros called from macros from another crate.